spec/sailfish_iptables_dbus_interface_description.xml

```diff
@@ -0,0 +1,464 @@
+</method>
+
+<!--
+Add IP address or network with port range (e.g., 2000:3000,) to filter table OUTPUT chain 
+as DROP.
+-->
+<method name="DenyOutgoingIpWithPortRange">
+<arg name="ip" type="s" direction="in"/>
+<arg name="port" type="s" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Add port with address any (0.0.0.0) to filter table OUTPUT chain as DROP.
+Parameter protocol can be omitted if port is of some known service (/etc/services)
+-->
+<method name="DenyOutgoingPort">
+<arg name="port" type="q" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- Add port range (e.g., 2000:3000) with address any (0.0.0.0) to filter table
+OUTPUT chain as DROP.
+-->
+<method name="DenyOutgoingPortRange">
+<arg name="port" type="s" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- Add IP address or network with service by name (has to exist in 
+/etc/services) to filter table OUTPUT chain as DROP.
+Parameter protocol can be omitted, if service is found.
+-->
+<method name="DenyOutgoingIpWithService">
+<arg name="ip" type="s" direction="in"/>
+<arg name="service" type="s" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- Add service by name (has to exist in /etc/services) with address any
+(0.0.0.0) to filter table OUTPUT chain as DROP.
+Parameter protocol can be omitted, if service is found.
+-->
+<method name="DenyOutgoingService">
+<arg name="service" type="s" direction="in"/>
+<arg name="protocol" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- Method that returns sailfish iptables api version -->
+<method name="GetVersion">
+<arg name="version" type="i" direction="out"/>
+</method>
+
+<!--
+Sent when plugin is loaded into Connection Manager to notify that iptables can
+be managed using this DBUS API.
+-->
+<signal name="Initialize"></signal>
+
+<!--
+Sent when plugin is removed from Connection Manager.
+-->
+<signal name="Shutdown"></signal>
+
+<!-- Sent when LoadFirewallFromDisk succeeds -->
+<signal name="FirewallLoaded"></signal>
+
+<!-- Sent when SaveFirewallToDisk succeeds -->
+<signal name="FirewallSaved"></signal>
+
+<!-- Sent when ClearFirewall succeeds -->
+<signal name="FirewallCleared"></signal>
```

Comment:
What's `ClearFirewall`?

Reply:
Changed method name in a636c6bfbd986ba333f506dbd593fdc470c13aa7 . this is sent when firewall is cleared. Maybe unnecessary.